### PR TITLE
Set `TF_CPP_MIN_LOG_LEVEL=0` in `__init__.py`

### PR DESCRIPTION
**Pull request by igorts-git**
# Description

Set `TF_CPP_MIN_LOG_LEVEL=0` in `__init__.py` instead of `train.py`.

Setting this env variable after `import jax` is called has no effect, because the C++ logging system is already initialized.

The `jax/__init__.py` sets the default value of `TF_CPP_MIN_LOG_LEVEL` to "1" ([code pointer](https://github.com/jax-ml/jax/blob/8504c16763b952b37c60f260db6cea84cc8ef387/jax/__init__.py#L17)). Then it calls `_cloud_tpu_init()`, which also initializes the C++ logging system.

There are many useful C++ LOG(INFO) messages that are suppressed without `TF_CPP_MIN_LOG_LEVEL=0`. For example, the path of the created profile trace (when running with `profiler=xplane` flag).

FIXES: b/474109045

# Tests

Local run of train.py with `profile=xplane` to confirm that the trace path is printed.

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [X] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [X] I have necessary comments in my code, particularly in hard-to-understand areas.
- [X] I have run end-to-end tests tests and provided workload links above if applicable.
- [X] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: src/MaxText/__init__.py [hunk 1]
```diff
@@ -27,6 +27,12 @@
 
 from collections.abc import Sequence
 
+import os as _os
```
[the remark this answers is not in the document]
Comment (igorts-git) [reply src/MaxText/__init__.py]: Done

File: src/MaxText/__init__.py [hunk 2]
```diff
@@ -27,6 +27,12 @@
...
+# In order to have any effect on the C++ logging this has to be set before we import anything from jax.
+# When jax is imported, its `__init__.py` calls `cloud_tpu_init()`, which also initializes the C++ logger.
+_os.environ.setdefault("TF_CPP_MIN_LOG_LEVEL", "0")
```
[the remark this answers is not in the document]
Comment (igorts-git) [reply src/MaxText/__init__.py]: I could not figure out how to add a reliable unit test for this. The tests are located outside of the "src/MaxText" directory and invoke `train.main()` as a function. This means that the order of imports in the test itself would affect the C++ logging behavior. Specifically, if the test imports JAX before "MaxText.train" the logging won't work.
Additionally, it is not easy to figure out how to capture C++ LOG(INFO) outputs in a unit tests. The standard Python unit testing libs only capture logs coming from Python.

**Review by gobbleturk — APPROVED** [review]
LGTM, why `del os` though?

**Comment by igorts-git** [comment]
> LGTM, why `del os` though?

I am not a Python expert, but I understand that this is the common convention to avoid pollution of the namespace. Without this `del` statement one can refer to the module os as `MaxText.os`, which would be confusing. FWIW, Gemini seems to agree with me that this is a nice thing to have.
